tests/integrations/base.py

```diff
@@ -104,3 +110,68 @@ def assert_new_commit(self, steps=1):
     def assert_file_content(self, file_path, content):
         with open(self.repo_path + "/" + file_path) as f:
             assert f.read() == content
+
+
+class GitFSLog(object):
+    def __init__(self, file_descriptor):
+        self._partial_line = None
+        self.line_buffer = collections.deque()
+        self.file_descriptor = file_descriptor
+
+    def get_line(self):
+        if not self.line_buffer:
+            self._read_data()
+        if self.line_buffer:
+            return self.line_buffer.popleft()
+
+    def _read_data(self):
+        data = os.read(self.file_descriptor, 2048).splitlines(True)
+        if not data:
+            return
+        if self._partial_line:
+            data[0] = self._partial_line + data[0]
+        if not data[-1].endswith("\n"):
+            self._partial_line = data[-1]
+            data = data[:-1]  # discard the partial line
+        else:
+            self._partial_line = None
+        self.line_buffer.extend(data)
+
+    def clear(self):
+        while os.read(self.file_descriptor, 2048):
+            pass
+        self._partial_line = None
+        self.line_buffer = collections.deque()
+
+    def __call__(self, expected, **kwargs):
+        @contextmanager
+        def log_context(gitfs_log):
+            gitfs_log.clear()
+            yield
+            if isinstance(expected, basestring):
+                gitfs_log.expect(expected, **kwargs)
+            else:
+                gitfs_log.expect_multiple(expected, **kwargs)
+        return log_context(self)
+
+    def expect(self, expected, timeout=10, pollfreq=0.1):
```

Comment:
I think that all this blocking logic should be moved in `get_line(self, timeout=10)`